src/pages/youtube-test/index.tsx

```diff
@@ -16,10 +30,41 @@ const YoutubeTestPage: NextComponentType = () => {
     alert(JSON.stringify(result));
   }, []);
 
+  const handleSearchQueryChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      setSearchQuery(event.target.value);
+    },
+    []
+  );
+
+  useDebounce(
+    async () => {
+      if (searchQuery === '') {
```

Comment:
falsy한 값체크는 
```suggestion
      if (!searchQuery) {
```
위와 같이 체크하는게 더 좋지 않을까요!

Reply:
빈 문자열만 체크하는 것이 자바스크립트의 이상한 falsy 판단을 피해갈 수 있어서 더 명확하다고 생각이 드는데, 그렇게 생각한 이유가 있을까요?

Reply:
예전에 모던 자바스크립트에서는 falsy 값의 체크는 저렇게 하는게 더 좋다구 들어서유..! 
이상한 falsy라면 혹시 어떤 경우를 말하는 걸까유?!

falsy라면 사실 위 경우에서는 어떤 경우라도 return이 낫지 않을까유!  물론 문자열만 넘어오겠지만유..!

Reply:
conditional statement 에 boolean이 아닌 다른 값을 넣을 경우, 의도치 않은 상황이 발생할 수 있습니다.

예)

```
// bad
function isNumberExist(numberOfUndefined: number | undefined) {
  return !!numberOfUndefined; // numberOfUndefined가 0일 경우에도 false를 리턴
}

// good
function isNumberExist(numberOfUndefined: number | undefined) {
  return numberOfUndefined !== undefined;
}


// soso
function isEmpty(text: string) {
  return !text;
}

// explicit
function isEmpty(text: string) {
  return text === '' // 또는, text.length > 0;
}
```

javascript에서는 모든 인자와 변수값에 null과 undefined가 들어올 가능성이 있기 때문에 방어적으로 짜기 위해 conditional statement에 boolean이 아닌 statement를 쓰는 것이 좋을 수 있는데요. 변수의 타입을 알 수 있는 typescript에서는 nil 값에 대한 체크는 이미 타입을 통해 검사되기  때문에, 조건문에서는 명확하게 검사하고자 하는 부분만 하는 것이 좋다고 생각이 듭니다.

참고로, boolean이 아닌 표현을 conditional statement에서 사용하지 못하게 제한하는 tslint rule도 존재합니다 (https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md)


다만, 언제나 이것이 옳은 것이 아니고 개발자들끼리 얼라인을 맞추고 결정해야하는 문제라 여러 사람의 의견을 구합니다

@Sabi-wa @kang-heewon

Reply:
저도 타입스크립트가 타입 체킹을 해주니 체크할 부분만 체크를 하면 좋을 것 같습니다!